Configurations/10-main.conf

```diff
@@ -477,7 +477,7 @@ my %targets = (
         CC               => "gcc",
         CFLAGS           => combine(picker(debug   => "-O0 -g",
                                            release => "-O3")),
-        cppflags         => threads("-D_REENTRANT"),
+        cflags           => add(threads("-pthread")),
```

Comment:
Side note: At least the GNU folks consider `-pthread` a cpp flag.
Ref: https://gcc.gnu.org/onlinedocs/gcc/Preprocessor-Options.html

Reply:
... and a linker flag as well, of course...
Ref: https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html

So strictly speaking, it should be placed in `cppflags` as well as `lflags`

Reply:
We use cflags used for both compiling and linking, so it doesn't *have to* be split. Besides, those online docs do say "You should use this option *consistently* for both compilation and linking." So keeping them in cflags is not actually inappropriate, as it formally decreases the chances of getting it wrong. And if you look closely, this fixes a problem with one of the targets, and problem is that -pthreads was used inconsistently.

Reply:
And keep in mind that -pthread is something you pass to [g]cc, *never* to ld. [g]cc translates it to suitable option for ld, so it *is* a "cflag" in every possible meaning.

Reply:
Well, considering [g]cc is a frontend to everything C, I understand the docs more as options for different aspects of building a program.  With that in mind, cppflags is for using [g]cc to preprocess (which necessarely also happens as part of compiling C source to an object file as well), cflags is for using [g]cc to compile C source to an object file, and ldflags is for using [g]cc to link object files into programs.

So no, strictly speaking, I disagree, and considering how the GNU folks have compartmentalised this option into preprocessor options *and* link options (their [summary](https://gcc.gnu.org/onlinedocs/gcc/Option-Summary.html) is pretty clear), it would be just as appropriate to have it as part of `cppflags` and `lflags`.

But note that this is a *side note*.  I don't intend to block this PR, but will probably formalise these thoughts some time in the future.

Reply:
At the same time we already established that there is no clear cut between preprocessor options and ones that directly affect code generation. Or in more general terms that there are options that would be wrong to specify at one stage and omit at another, even though they can be viewed as defining only one stage.  As ultimate example consider -fopenmp. What is it? Does the fact that it's listed in "C Language Options" mean that you can omit it when pre-processing? Linking? No and no. (And by the way, it does all the things -pthread does.)

And I don't see that the referred summary actually puts any obligation on you to assign options to specific or separate make variables. For example there are "C Language Options" and "Optimization Options". So following suggested logic [that presented classification imposes flag separation at make variables' level] you would have to limit CFLAGS to hold only former options and introduce some other flag to hold latter ones. And you would have to have another one to hold "Warning Options." And[!] you'd have to keep -D and -I options apart, because -D is listed in "Preprocessor Options" and -I in "Directory Options". To summarize. It's up to us how to handle options in our Makefile, and it's not inappropriate to minimize amount of variables. Rather contrary actually, it's not exactly appropriate to maximize amount of variables/options/joints...

Reply:
Or to put a little bit more practical spin on it. Since we *do* pass cflags at *all* stages, it's only appropriate to assign multi-stage flags [such as -pthread] to cflags. Viewing it from another angle, only pure single-stage flags [such as -L ,-l, -I, -D] are appropriate to separate as make variables. And to variables specific to corresponding stage, not some other grouping. Last remark is reference to the fact and if one follows suggested logic [that presented classification imposes specific separation as make variables], you'd have to pass -L at pre-processor stage and -I at linker, because they both are listed in "Directory Options".

Reply:
Speaking of appropriateness, I think it's *in*appropriate to continue derailing this PR, so I won't.  This discussion belongs in openssl-project, and I'll happily take it there.